test/phoenix/channel_test.exs

```diff
@@ -157,7 +158,7 @@ defmodule Phoenix.ChannelTest do
   end
 
   test "#broadcast_from and #broadcast_from! raises error when msg isn't a Map" do
-    socket = Socket.put_topic(new_socket, "top:subtop")
+    socket = new_socket |> put_in([:joined], true) |> Socket.put_topic("top:subtop")
```

Comment:
`new_socket |> put_in([:joined], true)` is used fairly often. Maybe extract a function called `new_joined_socket`?